File: PAP2324/test/tp01/TestMyMath.java
package tp01;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;
import static org.junit.jupiter.api.Assertions.assertThrows;

import org.junit.jupiter.api.Test;

class TestMyMath {
	
	@Test
	void testEstPair() {
		assertTrue(MyMath.estPair(8));
		assertFalse(MyMath.estPair(3));
		assertTrue(MyMath.estPair(0));
		assertThrows(AssertionError.class, () -> MyMath.estPair(-2));
	}
	
	@Test
	void testnbBitsV1() {
		assertEquals(0,MyMath.nbBitsV1(0));
		assertEquals(1,MyMath.nbBitsV1(1));
		assertEquals(32,MyMath.nbBitsV1(-1));
	}
	
	@Test
	void testnbBitsV2() {
		assertEquals(0,MyMath.nbBitsV2(0));
		assertEquals(1,MyMath.nbBitsV2(1));
		assertEquals(3,MyMath.nbBitsV2(0b000011010));
		assertEquals(32,MyMath.nbBitsV2(-1));
	}
	
	
}

File: PAP2324/python/test_mymath.py
import unittest
import mymath

class TestMyMath(unittest.TestCase):
    
    def test_estPair(self):
        self.assertEqual(True,mymath.estPair(20))
        self.assertEqual(False,mymath.estPair(21))
        self.assertEqual(True,mymath.estPair(0))
        self.assertRaises(AssertionError,mymath.estPair,-1)
        self.assertRaises(AssertionError,mymath.estPair,5.0)
        self.assertRaises(AssertionError,mymath.estPair,"brol")
    
    def test_nbBitsV1(self):
        self.assertEqual(0,mymath.nbBitsV1(0))
        self.assertEqual(1,mymath.nbBitsV1(1))
        self.assertEqual(6,mymath.nbBitsV1(207))
        
if __name__ == "__main__":
    unittest.main()